modules/configuration/ajax/process.php

```diff
@@ -27,39 +27,91 @@
         if ($value == "") {
             $DB->delete('Config', array('ID' => $key));
         } else {
-            $DB->update(
-                'Config',
-                array('Value' => $value),
-                array('ID' => $key)
-            );
+            // if no duplicate value then do updating
+            if (duplicateExistsInDropdown($key, $value)) {
+                $DB->update(
+                    'Config',
+                    array('Value' => $value),
+                    array('ID' => $key)
+                );
+            } else {
+                   header("HTTP/1.1 303 Duplicate value for update");
+                   exit();
+            }
         }
     } else { //add new or remove
         $keySplit   = explode("-", $key);
         $valueSplit = explode("-", $value);
         if ($keySplit[0] == 'add') {
             if ($value !== "") {
-                $DB->insert(
-                    'Config',
-                    array(
-                     'ConfigID' => $keySplit[1],
-                     'Value'    => $value,
-                    )
-                );
+                if (countDuplicate($keySplit[1], $value) == '0') {
+                    $DB->insert(
+                        'Config',
+                        array(
+                         'ConfigID' => $keySplit[1],
+                         'Value'    => $value,
+                        )
+                    );
+                } else {
+                        header("HTTP/1.1 303 Duplicate value");
+                        exit();
+                }
             }
         } elseif ($valueSplit[0] == 'remove') {
             $DB->delete('Config', array('ID' => $valueSplit[1]));
         }
     }
 }
-
-
+/**
+ * Check Duplicate value
+ *
+ * @param string $key   The value of the key
+ * @param string $value The value of the value
+ *
+ * @return string $result
+ */
+function countDuplicate($key,$value)
+{
+       $DB     =& Database::singleton();
+       $result = $DB->pselectOne(
+           "SELECT count(*) FROM Config WHERE ConfigID =:ConfigID AND Value =:Value",
+           array(
+            ':ConfigID' => $key,
+            ':Value'    => $value,
+           )
+       );
+       return $result;
+}
+/**
+ * Check dropdown list Duplicate value
+ *
+ * @param string $id    The value of the id
+ * @param string $value The value of the value
+ *
+ * @return boolean return true if there is no Duplicate value
+ */
+function duplicateExistsInDropdown($id,$value)
+{
+       $DB       =& Database::singleton();
+       $ConfigID = $DB->pselectOne(
+           "SELECT ConfigID FROM Config WHERE ID =:ID",
+           array(':ID' => $id)
+       );
+       $IDBefore = $DB->pselectOne(
+           "SELECT ID FROM Config WHERE ConfigID =:ConfigID AND Value =:Value",
+           array(
+            ':ConfigID' => $ConfigID,
+            ':Value'    => $value,
+           )
+       );
+       return ($id == $IDBefore || $IDBefore == null);
```

Comment:
I don't think this is reliable. You seem to be relying on the order that MySQL is returning rows without an explicit `ORDER BY` clause.

Reply:
After running the patch, all the ConfigID and Value pairs are unique, so no need to using the ORDER  BY in the query. I think.